jetstream/maxtext/maxengine-server/Dockerfile

```diff
@@ -0,0 +1,34 @@
+# Ubuntu:22.04
+# Use Ubuntu 22.04 from Docker Hub.
+# https://hub.docker.com/_/ubuntu/tags?page=1&name=22.04
+ARG UBUNTU_VERSION=83f0c2a8d6f266d687d55b5cb1cb2201148eb7ac449e4202d9646b9083f1cee0
+
+FROM ubuntu@sha256:${UBUNTU_VERSION} as base_image
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt -y update && apt install -y --no-install-recommends \
+    ca-certificates \
+    git \
+    python3.10 \
+    python3-pip
+
+RUN update-alternatives --install \
+    /usr/bin/python3 python3 /usr/bin/python3.10 1
+
+RUN git clone https://github.com/google/maxtext.git && \
+git clone https://github.com/google/JetStream.git
+
+RUN cd maxtext/ && \
+bash setup.sh
+
+RUN cd /JetStream && \
+pip install -e .
+
+RUN echo '#!/bin/bash\n\
```

Comment:
Would be good to create a separate entrypoint file with this script and copy it over instead of inlining since this is more error prone with spacing and future changes.

Reply:
Could you elaborate why it's more error prone? thanks!

Reply:
It is easier to make a mistake with adding escape sequences like `\n` where we miss a \ or misplace it leading to issues. Indentation is another problem where it is possible to mess up the indentation when you echo into a file. It is much cleaner to define the script outside and copy it in the Dockerfile using `COPY`. It is also easier to extend in the future if we need more changes.

Reply:
the points you mentioned are very common, any single character goes wrong could make things not work, for now, I don't think we need to spend the effort to do that

Reply:
Ok, I will make this as a potential enhancement, we could iterate in it later when have more cycles, thanks so much!